packages/router/src/shared.ts

```diff
@@ -156,3 +156,23 @@ export function defaultUrlMatcher(
 
   return {consumed: segments.slice(0, parts.length), posParams};
 }
+
+/**
+ * Takes two values and returns true only if they are the same value or are
+ * objects with the same properties whose values are also equal
+ * @param params two objects or string to deep compare.
+ * @returns true or false
+ */
+export function deepEqual(a: any, b: any) {
```

Comment:
```suggestion
export function shallowEqual(a: any, b: any) {
```

This is a shallow equality check, not deep.

Reply:
Hi @atscott,
I am a little bit confused here. Below is the definition of two types of checks:
Shallow equality checking (or reference equality) is checking that two different variables reference the same object; in contrast, deep equality checking (or value equality) is checking every value of two objects' properties.

And Isn't our code below checking every value in the object? 
```
aKeys.every(k => b.hasOwnProperty(k) && Object.is(a[k], b[k]));
```

Reply:
I guess this is one of those things that isn't exactly agreed upon by everyone. I think the argument would be that this is actually neither, but a [depth-1 equality check](https://books.google.com/books?id=CFtuCQAAQBAJ&pg=PA235&lpg=PA235&dq=shallow+equality+check+definition&source=bl&ots=9ip6ZiRSYE&sig=ACfU3U2p3Ef1jTj9WgFAISaQTXXhKkRvCw&hl=en&sa=X&ved=2ahUKEwjrmr_s0obqAhUKo54KHXndDToQ6AEwD3oECAgQAQ#v=onepage&q=shallow%20equality%20check%20definition&f=false). It's certainly not deep, which would imply that we recursively check the properties. But I can see that shallow equals might actually refer to reference equality.

Reply:
Agree with the depth-1 equality check! Should I name it depthOneEqual? or keep it as shallowEqual?

Reply:
depthOneEqual sounds good to me 👍